src/polycap-profile.c

```diff
@@ -0,0 +1,61 @@
+// get a new profile for a given type with properties
+polycap_profile* polycap_profile_new(
+	enum polycap_profile_type type,
+	double length,
+	double rad_ext[2],
+	double rad_int[2],
+	double focal_dist[2])
+{
+	struct cap_prof_arrays *shape_arr;
+
+	shape_arr = def_cap_profile(enum polycap_profile_type type, double length, double rad_ext[2], double rad_int[2], double focal_dist[2]);
+
+	return shape_arr;
```

Comment:
`struct cap_prof_arrays` en `polycap_profile` zijn wel niet hetzelfde he. 

Je call naar `def_cap_profile` is trouwens ook verkeerd met al die variabele types daarin.

Probeer eens dit alles te compileren (en al de errors te fixen 😄 ) door deze file toe te voegen aan `libpolycap_la_SOURCES` in `src/Makefile.am`.

Reply:
> struct cap_prof_arrays en polycap_profile zijn wel niet hetzelfde he.

Nee, maar kvind nergens waar polycap_profile werd gedefinieerd (althans niet met welke variabelen er in de structure zitten etc) dus dacht dat ik dat zelf mocht invullen. Leek me dat de reeds bestaande cap_prof_arrays struct dit reeds omvatte...

> Je call naar def_cap_profile is trouwens ook verkeerd met al die variabele types daarin.

Lijkt me dat enkel die enum variabele verkeerd is. Moet idd nog eens opzoeken hoe die enum precies werkt, maar kdacht dat type al direct de corresponderende integer ging bevatten, als je bijvoorbeeld polycap_profile_type(POLYCAP_PROFILE_PARABOLOIDAL) invoerde dat output '1' ging zijn.

Zal in namiddag wat proberen debuggen :)

Reply:
De bedoeling is dat jij zelf `struct _polycap_profile` etc gaat definieren in `src/polycap-private.h` en er in smijt wat je nodig hebt. Deze zullen dus overeenkomen of in geval lijken op sommige van de huidige structs die in `src/polycap.c` gebruikt worden. De bedoeling is dat de huidige (van Laszlo en jezelf) eruit gaan ten voordele van de nieuwe.

Bij die `def_cap_profile` call heb je de variabele types daar nog staan, wat onmogelijk is. Je hebt gewoon de functie declaratie geplakt daar 😄

Reply:
Oh en trouwens morgen vertrek ik voor twee weken op reis naar India. Ik zal uiteraard beschikbaar blijven voor hulp, maar misschien zal mijn reactie snelheid iets lager zijn, voornamelijk door slechte internet verbindingen...

Reply:
No probs. Kdenk dat ik wel even voort kan, tegen dat al die dingen heb geregeld :)
Mega veel plezier in India!

Reply:
Thanks!

Wanneer we klaar zijn met de release van de eerste versie zal je zien dat er van Laszlo's oorspronkelijke code nog weinig overblijft 😄

Reply:
Maar concreet herschrijf ik dan best ook altijd de inhoud van bvb def_cap_profile() ipv in de functie polycap_profile_new() daar gewoon gebruik van te maken? (mss niet zo kritiek bij dit voorbeeld, gezien hoe 'net' de laatste al is)
Want anders blijft Laszlo's code natuurlijk wel gebruikt worden, zij het gewoon gelinkt via andere functies?

Reply:
Het is best wat in def_cap_profile zit te brengen naar de implementatie in polycap-profile.c. Op termijn zou alles in polycap.c moeten herschreven worden naar de individuele polycap-*.c bestanden. Dit is vooral om onderhoud te vergemakkelijken, en ook voor potentiele samenwerkingen later.

Reply:
Alrighty :) No problem :)